dso-l1/src/main/java/com/tc/object/EntityClientEndpointImpl.java

```diff
@@ -235,6 +241,28 @@ public void close() {
     this.isOpen = false;
   }
 
+  @Override
+  public Future<Void> release() {
+    Callable<Void> call = new Callable<Void>() {
+      @Override
+      public Void call() throws Exception {
+        close();
+        return null;
+      }
+    };
+    if (this.closer == null) {
+      close();
+      return new CompletedFuture();
+    } else {
+      try {
+        return this.closer.submit(call);
+      } catch (RejectedExecutionException re) {
+        // connection already shutdown
+        return new CompletedFuture(new IllegalStateException("connection has already been shutdown"));
```

Comment:
with the choice of the unbounded executor service, I believe this is always true.